### PR TITLE
chore: use cross for linux-arm builds

### PR DESCRIPTION
**Pull request by 0xbe7a**
# Motivation

ubuntu-arm runners are not available in public repos

# Changes

Use cargo-cross to cross-compile for linux-aarch

### CODE REVIEW

**Comment by 0xbe7a** [comment]
Builds are now succeeding. I have used cross for all linux builds @pavelzw

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -33,18 +33,25 @@ jobs:
...
+            os: ubuntu-latest
+            cross: true
           - target: x86_64-pc-windows-msvc
             os: windows-latest
+            cross: false
           - target: aarch64-apple-darwin
             os: macos-latest
+            cross: false
           - target: x86_64-apple-darwin
             os: macos-13
```
Comment (pavelzw) [inline .github/workflows/build.yml:53]: at some point we need to use cross-compilation for osx-64 as well once github drops support for that... but maybe we can just stop supporting macos-64 at that point

**Review by pavelzw — APPROVED** [review]

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -33,18 +33,25 @@ jobs:
         include:
           - target: x86_64-unknown-linux-musl
             os: ubuntu-latest
+            cross: true
```
[the remark this answers is not in the document]
Comment (0xbe7a) [reply .github/workflows/build.yml:36]: Having a working cross-configuration, which includes a complete and reproducible non-rustc toolchain is preferable from a developer's standpoint as it makes debugging and testing easier. Cross can still make use of caching, so I don't think there are any major downsides to using it. Additionally, our CI ensures that the cross-configurations remain functional.

File: .github/workflows/build.yml (hunk 1 above)
Comment (0xbe7a) [reply .github/workflows/build.yml:53]: Cross does not yet have official arm64 images, but there are some community builds we could use once GitHub drops their runner support.

**Pull request merged**